### PR TITLE
[QAT-775] Book List & Details

### PR DESCRIPTION
**Pull request by ranuccimariano**
## Summary
- Added `books_list.feature` file to describe listing scenarios for wBooks API
    - Background:
        - added a backgroud so the user is logged in every scenario
    - Scenarios:
        - Accessing the books list
        - Access a book card to check its details
        - Access a book card to check the suggestions bar

## Jira Card
https://wolox-support.atlassian.net/browse/QAT-775?atlOrigin=eyJpIjoiYTc4MzVjMTYyNjJkNDU5ZjgxMGYwZDM2MjVjMWU2MjYiLCJwIjoiaiJ9

### CODE REVIEW

File: features/step_definitions/books.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+When('I click on any of the cards') do
+  clicked_book = @book_list.first
+  book = prepare_book(clicked_book)
+  @clicked_book_title = book['title'].text
+  @clicked_book_author = book['author'].text
+  click(:id, clicked_book.attribute(:id))
+end
+
+Then('I will be redirected to the book details') do
+  author = get_element_text(:id, 'deatil_author').slice(17..)
```
Comment (ffasitella) [inline features/step_definitions/books.rb]: Although this works correctly, I think using methods like `.sub('Autor del libro: ', '')` or `delete_prefix('Autor del libro')` would be easier to understand.

File: features/step_definitions/books.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
+Given('I have access to the book list') do
+  wait_for_element_to_be_ready(:id, 'book_list')
+  wait(1)
```
Comment (ffasitella) [inline features/step_definitions/books.rb]: Why do you need this?

File: features/step_definitions/books.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+And('I can check the book attributes') do
+  details_ids = %w[book_tiitle deatil_author detail_publisher detail_year]
+  details_ids.each do |detail_id|
+    expect(check_element_presence(:id, detail_id, true)).to be(true)
+  end
+  expect(check_element_presence(:class, 'genre', true)).to be(true)
+end
+
+Then('I can check the suggestions bar shows four results') do
```
Comment (ffasitella) [inline features/step_definitions/books.rb]: Use `And` instead of `Then` (just for convention)

File: features/step_definitions/books.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  wait_for_element_to_be_ready(:class, 'carousel-inner')
+  @book_slides = $driver.find_elements(:id, 'slide')
+  @book_slides_activos = process_slides(@book_slides)
+  expect(@book_slides_activos.length).to eq(4)
+end
+
+Then('I can navigate {string} when i press the {string} button') do |_dir, direction|
```
Comment (ffasitella) [inline features/step_definitions/books.rb]: Same as above.
Also, I like this implementation :+1:.

File: features/books_list.feature [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  Background: I am logged in
+    Given I am on the login page
+    And I fill the required fields of the login form
+    And I click on the login button
+    And I will "redirect to" the "books" page
+
+  Scenario: Access to books list
+    Given I have access to the book list
+    When All the books are displayed
+    Then I can check all the books have a "title" attribute
```
Comment (guerragomez) [inline features/books_list.feature]: in this case it would be good to apply the outline scenarios

File: features/books_list.feature [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    Then I will be redirected to the book details
+    And I can check the book attributes
+
+  Scenario: Navigate suggestions bar
+    Given I have access to the book list
+    When All the books are displayed
+    And I click on any of the cards
+    Then I will be redirected to the book details
+    And I can check the suggestions bar shows four results
+    And I can navigate "right" when i press the "next" button
```
Comment (guerragomez) [inline features/books_list.feature]: in this case it would be good to apply the outline scenarios

File: features/step_definitions/books.rb (hunk 2 above)
Comment (guerragomez) [reply features/step_definitions/books.rb]: you can implement this action in another step

File: features/step_definitions/books.rb (hunk 4 above)
Comment (ranuccimariano) [reply features/step_definitions/books.rb]: Thanks :smile:

File: features/step_definitions/books.rb (hunk 3 above)
Comment (ranuccimariano) [reply features/step_definitions/books.rb]: Sorry i forgot :sweat_smile:

File: features/step_definitions/books.rb (hunk 2 above)
Comment (ranuccimariano) [reply features/step_definitions/books.rb]: if i don't use the wait, the books don't have enough time to load so i can't create the array with `find_elements` (the array is empty). I know it's not the right way but i couldn't find an explicit wait that fits

File: features/step_definitions/books.rb (hunk 1 above)
Comment (ranuccimariano) [reply features/step_definitions/books.rb]: Thanks! i didn't know that method :sweat_smile:

**Pull request merged**